src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,37 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0393: r##"
+A type parameter which references `Self` in its default value was not specified.
+Example of erroneous code:
```

Comment:
I don't currently have an opinion about whether the phrase in question needs to take the form of a sentence or not, but I do have an opinion about the supposedly more natural alternative of "Erroneous code example":

The phrase "Example of erroneous code" is at least unambiguous about what is erroneous: the code itself.

The phrase "Erroneous code example" is less clear.
 * The adjective "Erroneous" could be interpreted as applying to the word "code", as in:

    > here is some code that illustrates the error under discussion

 * or it could alternatively be interpreted as applying to the word "example", as in: 

    > here is an attempt at an example of some code to illustrate concept X, but the example is erroneous, and thus the example fails to actually illustrate X

(Of course one has to actually *work* to come up with the latter interpretation. Nonetheless, I'm generally -1 on phrasing that slip too easily into multiple potential parsings.)